src/chess/board.h

```diff
@@ -81,8 +81,10 @@ class ChessBoard {
   uint64_t Hash() const {
     return HashCat({our_pieces_.as_int(), their_pieces_.as_int(),
                     rooks_.as_int(), bishops_.as_int(), pawns_.as_int(),
-                    our_king_.as_int(), their_king_.as_int(),
-                    castlings_.as_int(), flipped_});
+                    (static_cast<unsigned>(our_king_.as_int()) << 24) |
```

Comment:
`unsigned` has different length at different platforms, it's safer to cast to something of known size (e.g. `uint64_t` or `uint32_t`)

Reply:
Thanks, you are right!